# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rocm-llvm/gcc-path.patch
# Copyright (C) 2021 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- clang-14.0.0.src/clang/lib/Driver/ToolChains/Gnu.cpp.vanilla	2022-03-14 10:44:55.000000000 +0100
+++ clang-14.0.0.src/clang/lib/Driver/ToolChains/Gnu.cpp	2022-03-26 09:37:53.569047800 +0100
@@ -2074,10 +2074,10 @@
   static const char *const AArch64Triples[] = {
-      "aarch64-none-linux-gnu", "aarch64-linux-gnu", "aarch64-redhat-linux",
-      "aarch64-suse-linux"};
+      "aarch64-t2-linux-gnu", "aarch64-t2-linux-musl"};
   static const char *const AArch64beLibDirs[] = {"/lib"};
-  static const char *const AArch64beTriples[] = {"aarch64_be-none-linux-gnu",
-                                                 "aarch64_be-linux-gnu"};
+  static const char *const AArch64beTriples[] = {"aarch64_be-t2-linux-gnu",
+                                                 "aarch64_be-t2-linux-musl"};
 
   static const char *const ARMLibDirs[] = {"/lib"};
-  static const char *const ARMTriples[] = {"arm-linux-gnueabi"};
+  static const char *const ARMTriples[] = {"arm-t2-linux-gnueabi",
+	                                   "arm-t2-linux-musl"};
   static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
@@ -2096,8 +2096,3 @@
   static const char *const X86_64Triples[] = {
-      "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
-      "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
-      "x86_64-redhat-linux",    "x86_64-suse-linux",
-      "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
-      "x86_64-slackware-linux", "x86_64-unknown-linux",
-      "x86_64-amazon-linux"};
+      "x86_64-t2-linux-gnu",    "x86_64-t2-linux-musl"};
   static const char *const X32Triples[] = {"x86_64-linux-gnux32",
@@ -2483,9 +2483,10 @@
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
-      "i586-linux-gnu",      "i686-linux-gnu",        "i686-pc-linux-gnu",
-      "i386-redhat-linux6E", "i686-redhat-linux",     "i386-redhat-linux",
-      "i586-suse-linux",     "i686-montavista-linux",
+      "i386-t2-linux-gnu",  "i486-t2-linux-gnu",  "i586-t2-linux-gnu",
+      "i686-t2-linux-gnu",
+      "i386-t2-linux-musl", "i486-t2-linux-musl", "i586-t2-linux-musl",
+      "i686-t2-linux-musl"
   };
 
   static const char *const LoongArch64LibDirs[] = {"/lib64", "/lib"};
@@ -2118,7 +2114,6 @@
   static const char *const MIPSTriples[] = {
-      "mips-linux-gnu", "mips-mti-linux", "mips-mti-linux-gnu",
-      "mips-img-linux-gnu", "mipsisa32r6-linux-gnu"};
+      "mips-t2-linux-gnu", "mips-t2-linux-musl"};
   static const char *const MIPSELLibDirs[] = {"/libo32", "/lib"};
   static const char *const MIPSELTriples[] = {
-      "mipsel-linux-gnu", "mips-img-linux-gnu", "mipsisa32r6el-linux-gnu"};
+      "mipsel-t2-linux-gnu", "mipsel-t2-linux-musl"};
 
@@ -2126,10 +2121,6 @@
   static const char *const MIPS64Triples[] = {
-      "mips64-linux-gnu",      "mips-mti-linux-gnu",
-      "mips-img-linux-gnu",    "mips64-linux-gnuabi64",
-      "mipsisa64r6-linux-gnu", "mipsisa64r6-linux-gnuabi64"};
+      "mips64-t2-linux-gnuabi64", "mips64-t2-linux-muslabi64"};
   static const char *const MIPS64ELLibDirs[] = {"/lib64", "/lib"};
   static const char *const MIPS64ELTriples[] = {
-      "mips64el-linux-gnu",      "mips-mti-linux-gnu",
-      "mips-img-linux-gnu",      "mips64el-linux-gnuabi64",
-      "mipsisa64r6el-linux-gnu", "mipsisa64r6el-linux-gnuabi64"};
+      "mips64el-t2-linux-gnuabi64", "mips64el-t2-linux-muslabi64"};
 
@@ -2134,4 +2134,4 @@
   static const char *const MIPSN32LibDirs[] = {"/lib32"};
-  static const char *const MIPSN32Triples[] = {"mips64-linux-gnuabin32",
-                                               "mipsisa64r6-linux-gnuabin32"};
+  static const char *const MIPSN32Triples[] = {"mips64-t2-linux-gnuabin32",
+                                               "mips64-t2-linux-muslabin32"};
   static const char *const MIPSN32ELLibDirs[] = {"/lib32"};
@@ -2140,3 +2131,3 @@
   static const char *const MIPSN32ELTriples[] = {
-      "mips64el-linux-gnuabin32", "mipsisa64r6el-linux-gnuabin32"};
+      "mips64el-t2-linux-gnuabin32", "mips64el-t2-linux-muslabin32"};
 
@@ -2147,10 +2138,8 @@
   static const char *const PPCTriples[] = {
-      "powerpc-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc-linux-gnuspe",
       // On 32-bit PowerPC systems running SUSE Linux, gcc is configured as a
       // 64-bit compiler which defaults to "-m32", hence "powerpc64-suse-linux".
-      "powerpc64-suse-linux", "powerpc-montavista-linuxspe"};
+      "powerpc-t2-linux-gnu", "powerpc-t2-linux-musl"};
   static const char *const PPCLELibDirs[] = {"/lib32", "/lib"};
-  static const char *const PPCLETriples[] = {"powerpcle-linux-gnu",
-                                             "powerpcle-unknown-linux-gnu",
-                                             "powerpcle-linux-musl"};
+  static const char *const PPCLETriples[] = {"powerpcle-t2-linux-gnu",
+                                             "powerpcle-t2-linux-musl"};
 
@@ -2158,25 +2147,20 @@
   static const char *const PPC64Triples[] = {
-      "powerpc64-linux-gnu", "powerpc64-unknown-linux-gnu",
-      "powerpc64-suse-linux", "ppc64-redhat-linux"};
+      "powerpc64-t2-linux-gnu", "powerpc64-t2-linux-musl"};
   static const char *const PPC64LELibDirs[] = {"/lib64", "/lib"};
   static const char *const PPC64LETriples[] = {
-      "powerpc64le-linux-gnu", "powerpc64le-unknown-linux-gnu",
-      "powerpc64le-none-linux-gnu", "powerpc64le-suse-linux",
-      "ppc64le-redhat-linux"};
+      "powerpc64le-t2-linux-gnu", "powerpc64le-t2-linux-musl"};
 
   static const char *const RISCV32LibDirs[] = {"/lib32", "/lib"};
-  static const char *const RISCV32Triples[] = {"riscv32-unknown-linux-gnu",
-                                               "riscv32-linux-gnu",
-                                               "riscv32-unknown-elf"};
+  static const char *const RISCV32Triples[] = {"riscv32-t2-linux-gnu",
+                                               "riscv32-t2-linux-musl"};
   static const char *const RISCV64LibDirs[] = {"/lib64", "/lib"};
-  static const char *const RISCV64Triples[] = {"riscv64-unknown-linux-gnu",
-                                               "riscv64-linux-gnu",
-                                               "riscv64-unknown-elf"};
+  static const char *const RISCV64Triples[] = {"riscv64-t2-linux-gnu",
+                                               "riscv64-t2-linux-musl"};
 
   static const char *const SPARCv8LibDirs[] = {"/lib32", "/lib"};
-  static const char *const SPARCv8Triples[] = {"sparc-linux-gnu",
-                                               "sparcv8-linux-gnu"};
+  static const char *const SPARCv8Triples[] = {"sparc-t2-linux-gnu", "sparcv9-t2-linux-gnu",
+                                               "sparc-t2-linux-musl", "sparcv9-t2-linux-musl"};
   static const char *const SPARCv9LibDirs[] = {"/lib64", "/lib"};
-  static const char *const SPARCv9Triples[] = {"sparc64-linux-gnu",
-                                               "sparcv9-linux-gnu"};
+  static const char *const SPARCv9Triples[] = {"sparc64-t2-linux-gnu",
+                                               "sparc64-t2-linux-musl"};
 
@@ -2184,4 +2168,3 @@
   static const char *const SystemZTriples[] = {
-      "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
-      "s390x-suse-linux", "s390x-redhat-linux"};
+      "s390x-t2-linux-gnu", "s390x-t2-linux-musl"};
 
